htdocs/compta/prelevement/create.php

```diff
@@ -203,20 +207,33 @@
 
 llxHeader('', $title);
 
-print load_fiche_titre($title);
 
-print dol_get_fiche_head();
+$head = bon_prelevement_prepare_head($bprev, $bprev->nbOfInvoiceToPay($type), $bprev->nbOfInvoiceToPay($type, 'salary'));
+if ($type) {
+	print dol_get_fiche_head($head, (!GETPOSTISSET('sourcetype') ? 'invoice' : 'salary'), $langs->trans("Invoices"), -1, $bprev->picto);
+} else {
+	print load_fiche_titre($title);
+	print dol_get_fiche_head();
+}
 
-$nb = $bprev->nbOfInvoiceToPay($type);
-$pricetowithdraw = $bprev->SommeAPrelever($type);
+
+if (!GETPOSTISSET('sourcetype')) {
+	$nb = $bprev->nbOfInvoiceToPay($type);
+	$pricetowithdraw = $bprev->SommeAPrelever($type);
+} else {
+	$nb = $bprev->nbOfInvoiceToPay($type, 'salary');
+	$pricetowithdraw = $bprev->SommeAPrelever($type, 'salary');
+}
 if ($nb < 0) {
 	dol_print_error($bprev->error);
 }
 print '<table class="border centpercent tableforfield">';
 
 $labeltoshow = $langs->trans("NbOfInvoiceToWithdraw");
-if ($type == 'bank-transfer') {
+if ($type == 'bank-transfer' && !GETPOSTISSET('sourcetype') || empty($type)) {
```

Comment:
The condition to know if this is a BonPrelevement for an invoice or a salary  is not clear.
It must be saved into 1 unique variable.
Also you mix && and or on the same line without parenthesis. Result is unpredicable.

Reply:
here I prefer to add brackets because this page is also shared by payment by direct debit when the type is empty